### DISCUSSION TITLE
Is Quasar.js supported?

### DISCUSSION THREAD

**Discussion by heliomar-pena** [fwouts/previewjs#654] [General]
Quasar.js have a library of global components, I want to know if I can use Preview.js with Quasar without problems. Currently, I'm receiving this error when I import a Quasar global component:

![image](https://user-images.githubusercontent.com/66505715/172492683-aa54d4e5-e554-4bfc-8aac-049b4862cefd.png)

**Comment by fwouts** [comment]
Hi @victor-heliomar, it should be possible with a little bit of configuration!

I haven't tried it but this might work:
```js
// FILE: preview.config.js
const { quasar } = require('@quasar/vite-plugin');

module.exports = {
  vite: {
    plugins: [
      quasar()
    ]
  }
}
```

**Reply by heliomar-pena** [reply]
Hi @fwouts , thanks for answering quickly. I have tested that configuration. It seems to have worked for some things, for example, if I import an icon from quasar, previewjs loads it. But it looks incomplete, for example, this component it's supposed to be a spinner, but on previewjs it looks like a circle, currently Quasar is not giving me an error, but the spinner is just a circle.

![LastRecord](https://user-images.githubusercontent.com/66505715/172496612-c45f8777-c9e5-492a-bb1e-ed57692707a5.gif)

Really it should look so:

![LastRecord](https://user-images.githubusercontent.com/66505715/172497508-76a84510-9493-457c-8979-e17dcb63ae86.gif)

However, it is progress!. I think a wrapper can help with this, I just need to get the CSS class of Quasar and put it on the Wrapper

---

That is about the components that just have images, but the components that handle logic, such as input, for example, are giving me this error:

![image](https://user-images.githubusercontent.com/66505715/172497637-955e3da5-1d71-4c19-8ade-99332a0adad3.png)

**Reply by heliomar-pena** [reply]
I have fixed the spinner problem by adding a wrapper with the Quasar CSS: 🎉

![LastRecord](https://user-images.githubusercontent.com/66505715/172498617-26ce079e-36e2-4285-8d40-bfa7379fcb7e.gif)

But I still don't know what can I do to solve this error 😥

![image](https://user-images.githubusercontent.com/66505715/172498509-918350ee-0e63-40be-b021-0cab2c16f927.png)

**Reply by fwouts** [reply]
Cool!

I think what you're probably missing is telling Vue to use Quasar, i.e. `app.use(Quasar)`.

Luckily someone else already figured out how to do this! ➡️ https://github.com/fwouts/previewjs/issues/113#issuecomment-1018250356

I guess it would look like (note I'm not familiar with Vue so syntax might be incorrect):
```vue
// Wrapper.vue
<script>
import Quasar from 'quasar';  // or maybe it's not the default export? I don't know :)

const {app} = getCurrentInstance()
app.use(Quasar);
</script>
<template>
  <slot></slot>
</template>
```

**Reply by heliomar-pena** [reply]
It is working perfectly for me 😌

![LastRecord](https://user-images.githubusercontent.com/66505715/172502652-2c225a18-3dfb-4b25-82aa-5b21dabb1eea.gif)

Now I just need to add the rest of the things like icons, but it's all I think 

![image](https://user-images.githubusercontent.com/66505715/172502776-219806fc-1f00-499b-8f3d-a1cadd5bd4e5.png)

Deeply grateful for your help
